specs/Standards/NonFungibleToken/Payout.md

```diff
@@ -54,44 +60,53 @@ Financial contracts MAY take a cut of the NFT sale price as commission, subtract
 ```
 
 ## Reference-level explanation
+
 ```rust
 /// A mapping of NEAR accounts to the amount each should be paid out, in
 /// the event of a token-sale. The payout mapping MUST be shorter than the
 /// maximum length specified by the financial contract obtaining this
 /// payout data. Any mapping of length 10 or less MUST be accepted by
 /// financial contracts, so 10 is a safe upper limit.
-#[derive(Serialize, Deserialize)]
+#[Derive(Serialize, Deserialize)]
 #[serde(crate = "near_sdk::serde")]
 pub struct Payout {
-  pub payout: HashMap<AccountId, U128>,
+  payout: HashMap<AccountId, U128>,
 }
 
-pub trait Payouts{
+pub trait Payouts {
   /// Given a `token_id` and NEAR-denominated balance, return the `Payout`.
   /// struct for the given token. Panic if the length of the payout exceeds
   /// `max_len_payout.`
-  fn nft_payout(&self, token_id: String, balance: U128, max_len_payout: u32) -> Payout;
+  fn nft_payout(&self, token_id: String, balance: U128, max_len_payout: Option<u32>) -> Payout;
   /// Given a `token_id` and NEAR-denominated balance, transfer the token
   /// and return the `Payout` struct for the given token. Panic if the
   /// length of the payout exceeds `max_len_payout.`
-  #[payable]
   fn nft_transfer_payout(
     &mut self,
     receiver_id: AccountId,
     token_id: String,
     approval_id: u64,
```

Comment:
was this supposed to be optional as well? as I see in this implementation:
https://github.com/TENK-DAO/tenk/blame/d65e15c5fdc4b70d6aed0c8181bce33ea0a16e68/contracts/tenk/src/payout.rs#L73

Reply:
```suggestion
    approval_id: Option<u64>,
```